org-code-javabuilder/theater/src/main/java/org/code/theater/Stage.java

```diff
@@ -386,4 +394,28 @@ private void drawImageHelper(
       this.graphics.drawImage(image, x, y, width, height, null);
     }
   }
+
+  private void writeImageAndAudioToFile() {
+    try {
+      String imageUrl =
+          this.fileWriter.writeToFile(
+              IMAGE_FILENAME, new ByteArrayInputStream(this.imageOutputStream.toByteArray()));
+      String audioUrl =
+          this.fileWriter.writeToFile(
+              AUDIO_FILENAME, new ByteArrayInputStream(this.audioOutputStream.toByteArray()));
+
+      HashMap<String, String> imageMessage = new HashMap<>();
+      imageMessage.put("url", imageUrl);
+      this.outputAdapter.sendMessage(new TheaterMessage(TheaterSignalKey.VISUAL_URL, imageMessage));
+
+      HashMap<String, String> audioMessage = new HashMap<>();
+      audioMessage.put("url", audioUrl);
+      this.outputAdapter.sendMessage(new TheaterMessage(TheaterSignalKey.AUDIO_URL, audioMessage));
+    } catch (JavabuilderException e) {
+      // we should not hit this (caused by too many file writes)
+      // in normal execution as it is only called via play,
+      // and play can only be called once.
```

Comment:
Given this, just a semantic question - should `TOO_MANY_WRITES` just be an internal exception then? Would a user be able to hit this case normally?

Reply:
answered in slack